### PR TITLE
Add API to enable or disable bare slash regex parsing

### PR DESCRIPTION
**Pull request by ahoppen**
Depends on https://github.com/apple/swift/pull/59018

---

Add a parameter to the `SyntaxParser.parse` functions to enable or disable bare slash regex parsing.

rdar://93750821

### CODE REVIEW

File: Sources/SwiftSyntaxParser/SyntaxParser.swift [hunk 1]
```diff
@@ -92,33 +104,55 @@ public enum SyntaxParser {
...
   ///       diagnostics the parser emits
   /// - Returns: A top-level Syntax node representing the contents of the tree,
   ///            if the parse was successful.
   /// - Throws: `ParserError`
-  public static func parse(_ url: URL,
-      diagnosticHandler: ((Diagnostic) -> Void)? = nil) throws -> SourceFileSyntax {
+  public static func parse(
+    _ url: URL,
+    languageVersion: String? = nil,
+    enableBareSlashRegexLiteral: Bool? = nil,
```
Comment (rintaro) [inline Sources/SwiftSyntaxParser/SyntaxParser.swift:121]: I *feel* like these should be a `Options` object. But I'm not sure how the API should look like, for example
```swift
// array of enums
SyntaxParse.parse(
    url,
    options: [.languageVersion("5.6"), .bareSlashRegexLiteral(true)],
    diagnosticHandler: diagnosticHandler)
// Just a "options" object
SyntaxParse.parse(
    url,
    options: SyntaxParseOptions(languageVersion: "5.1", enableBareSlashRegexLiteral: true),
    diagnosticHandler: diagnosticHandler)
```
The advantage of "options" object is that you can _easily_ pass around and store the object in the clients, when they want to create options at _not_  exact the call site.

WDYT?

Comment (rintaro) [reply Sources/SwiftSyntaxParser/SyntaxParser.swift:121]: Alex and I chatted about this and we decided to accept this as is for now until the "language features" proposal is settled.

**Review by rintaro — APPROVED** [review]
Just resolve the conflict :)

**Comment by ahoppen** [comment]
https://github.com/apple/swift/pull/59018

@swift-ci Please test

**Pull request merged**